unstructured/ingest/connector/airtable.py

```diff
@@ -51,8 +57,22 @@ class AirtableIngestDoc(IngestDocCleanupMixin, BaseIngestDoc):
 
     connector_config: SimpleAirtableConfig
     table_meta: AirtableTableMeta
+    retry_strategy_config: t.Optional[RetryStrategyConfig] = None
     registry_name: str = "airtable"
 
+    @property
+    def retry_handler(self) -> t.Optional[RetryHandler]:
+        if retry_strategy_config := self.retry_strategy_config:
+            return RetryHandler(
+                backoff.expo,
+                HTTPError,
+                max_time=retry_strategy_config.max_time,
+                max_tries=retry_strategy_config.max_tries,
+                logger=logger,
+                backoff_log_level=logging.DEBUG if self.processor_config.verbose else logging.INFO,
+            )
+        return None
+
```

Comment:
any way so that we don't have to copy and paste this to each connector?

Reply:
Added it to the pipeline rather than the doc itself, with a new custom error to know if something should be retried. Now it only needs to be instantiated once when the pipeline is created and used when by the source node when `doc.read_file` is called.